Netkan/Processors/QueueHandler.cs

```diff
@@ -0,0 +1,206 @@
+            if (msg != null)
+            {
+                log.Debug("Message received");
+                client.DeleteMessage(new DeleteMessageRequest()
+                {
+                    QueueUrl      = url,
+                    ReceiptHandle = msg.ReceiptHandle,
+                });
+            }
+            return msg;
+        }
+
+        private void sendCkan(Metadata ckan, Metadata netkan, bool success, string err = null)
+        {
+            var attribs = new Dictionary<string, MessageAttributeValue>()
+            {
+                {
+                    "ModIdentifier",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = netkan.Identifier
+                    }
+                },
+                {
+                    "Staged",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = netkan.Staged.ToString()
+                    }
+                },
+                {
+                    "Success",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = success.ToString()
+                    }
+                },
+                {
+                    "CheckTime",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = DateTime.UtcNow.ToString("s", CultureInfo.InvariantCulture)
+                    }
+                },
+            };
+            if (!string.IsNullOrEmpty(err))
+            {
+                attribs.Add(
+                    "ErrorMessage",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = err
+                    }
+                );
+            }            
+            
+            SendMessageRequest msg = new SendMessageRequest()
+            {
+                QueueUrl               = outputQueueURL,
+                MessageGroupId         = "1",
+                MessageDeduplicationId = Path.GetRandomFileName(),
+                MessageBody            = serializeCkan(ckan),
+                MessageAttributes      = attribs,
+            };
+            try
+            {
+                var resp = client.SendMessage(msg);
+            }
+            catch (Exception e)
+            {
+                log.ErrorFormat("Send failed: {0}\r\n{1}", e.Message, e.StackTrace);
+            }
+        }
+
+        private string serializeCkan(Metadata ckan)
+        {
+            if (ckan == null)
+            {
+                // SendMessage doesn't like empty bodies, so send an empty JSON object
+                return "{}";
+            }
```

Comment:
We could send the error message as a json object? The body is likely required as part of their queue deduplication efforts. I don't really have a strong opinion either way here.

Reply:
We could try something like that, but if our payload is defined as the .ckan JSON, then we'd be sending JSON that doesn't match that definition some of the time.

Or some kind of dual-purpose wrapper object that could contain either a .ckan or an error in the style of a C union, but then we're making up a JSON format that we don't have a schema for. And it starts to look like the outer layer of that object would be redundant with message attributes anyway.

```json
{
    "identifier": "AwesomeMod",
    "ckan": {
        "spec_version": 1,
        etc...
    },
    "error": "Didn't work"
}
```

Do you know how this sort of thing is handled in other queue setups? We can't be the only project with a process that may fail to generate the expected output.

Reply:
Yeah, I think you're right here. The body should be the body and essentially an error is a blank object, that way we're not overloading what it means.